### PR TITLE
Add NEMO regex

### PR DESCRIPTION
**Pull request by malmans2**
NEMO domain files often don't have attributes, I've added a few regex that should allow to guess most of the axes/coords:

- T, time: `t`, `time_{counter,instant,centered}`
- Z, vertical: `z`, `nav_lev`, `gdep{t,u,v,f,w,uw,vw}_{0,1d}`, `depth{t,u,v,f,w,uw,vw}`
- longitude: `nav_lon`, `glam{t,u,v,f}`
- latitude: `nav_lat`, `gphi{t,u,v,f}`

### CODE REVIEW

**Comment by dcherian** [comment]
Can you review & merge #172 and then add some tests here please?

File: cf_xarray/accessor.py [hunk 1]
```diff
@@ -116,15 +116,15 @@
 
 #: regular expressions for guess_coord_axis
 regex = {
-    "time": "time[0-9]*|min|hour|day|week|month|year",
+    "time": "\\bt\\b|time[0-9]*|min|hour|day|week|month|year",
     "vertical": (
-        "(lv_|bottom_top|sigma|h(ei)?ght|altitude|depth|isobaric|pres|"
-        "isotherm)[a-z_]*[0-9]*"
+        "\\bz\\b|(nav_)?(lev|gdep|lv_|bottom_top|sigma|h(ei)?ght|altitude|depth|"
```
Comment (dcherian) [inline cf_xarray/accessor.py:121]:
```suggestion
        "\\bz\\b|(nav_|lev|gdep|lv_|bottom_top|sigma|h(ei)?ght|altitude|depth|"
```
would this work?

File: cf_xarray/accessor.py [hunk 2]
```diff
@@ -116,15 +116,15 @@
...
+        "isobaric|pres|isotherm)[a-z_]*[0-9]*"
     ),
     "Y": "y",
-    "latitude": "y?lat[a-z0-9]*",
+    "latitude": "(y|nav_)?(lat|gphi)[a-z0-9]*",
     "X": "x",
-    "longitude": "x?lon[a-z0-9]*",
+    "longitude": "(x|nav_)?(lon|glam)[a-z0-9]*",
```
Comment (dcherian) [inline cf_xarray/accessor.py:127]:
```suggestion
    "longitude": "x?(|nav_|lon|glam)[a-z0-9]*",
```

File: cf_xarray/accessor.py [hunk 3]
```diff
@@ -116,15 +116,15 @@
...
+    "time": "\\bt\\b|time[0-9]*|min|hour|day|week|month|year",
     "vertical": (
-        "(lv_|bottom_top|sigma|h(ei)?ght|altitude|depth|isobaric|pres|"
-        "isotherm)[a-z_]*[0-9]*"
+        "\\bz\\b|(nav_)?(lev|gdep|lv_|bottom_top|sigma|h(ei)?ght|altitude|depth|"
+        "isobaric|pres|isotherm)[a-z_]*[0-9]*"
     ),
     "Y": "y",
-    "latitude": "y?lat[a-z0-9]*",
+    "latitude": "(y|nav_)?(lat|gphi)[a-z0-9]*",
```
Comment (dcherian) [inline cf_xarray/accessor.py:125]:
```suggestion
    "latitude": "y?(nav_|lat|gphi)[a-z0-9]*",
```

?

Comment (malmans2) [reply cf_xarray/accessor.py]: With this `"y?(nav_|lat|gphi)[a-z0-9]*"` I think nav_lon/lat/lev would conflict. Maybe let's keep it simple and use `"y?(nav_lat|lat|gphi)[a-z0-9]*"`

**Comment by malmans2** [comment]
Maybe for Z we can do the same as X and Y? So use `z?` rather than `\\bz\\b|`?
(E.g., MITgcm uses Z, Zp1, Zu, Zl)

**Comment by dcherian** [comment]
> Maybe for Z we can do the same as X and Y? So use z? rather than \\bz\\b|?
(E.g., MITgcm uses Z, Zp1, Zu, Zl)

Sounds good.

**Pull request merged**